common/components/controllers/CreateEventProjectController.jsx

```diff
@@ -0,0 +1,207 @@
+          formComponent: CreateEventProjectScope,
+        },
+        {
+          header: "What resources would you like to share?",
+          subHeader:
+            "Share your internal resources to help volunteers understand the goals and processes. " +
+            "Any changes below will not impact the details of your long-term project profile.",
+          onSubmit: this.onSubmit,
+          onSubmitSuccess: this.onNextPageSuccess,
+          formComponent: EventProjectResourcesForm,
+        },
+        {
+          header: "What type of volunteers does your hackathon team need?",
+          subHeader:
+            "You can always change the type of help your team needs later. " +
+            "Any changes below will not impact the details of your long-term project profile.",
+          submitButtonText: "Submit",
+          onSubmit: this.onSubmit,
+          onSubmitSuccess: this.onFinalSubmitSuccess,
+          formComponent: EventProjectPositionsForm,
+        },
+      ],
+    };
+  }
+
+  static getStores(): $ReadOnlyArray<FluxReduceStore> {
+    return [FormFieldsStore];
+  }
+
+  static calculateState(prevState: State, props: Props): State {
+    let state: State = _.clone(prevState) || {};
+    const project: MyProjectData = FormFieldsStore.getFormFieldValue("project");
+    if (project) {
+      state.projectId = project?.project_id;
+    }
+    return state;
+  }
+
+  componentDidMount(): void {
+    if (this.state.projectId) {
+      this.setState(
+        { isLoading: true },
+        EventProjectAPIUtils.fetchEventProjectDetails(
+          this.state.eventId,
+          this.state.projectId,
+          this.loadEventProjectDetails.bind(this),
+          this.handleLoadProjectError.bind(this)
+        )
+      );
+    }
+  }
+  updatePageUrl() {
+    if (this.state.projectId && !url.argument("project_id")) {
+      url.updateArgs({ project_id: this.state.projectId });
+    }
+    utils.navigateToTopOfPage();
+  }
+
+  loadEventProjectDetails(eventProject: EventProjectAPIDetails): void {
+    if (!CurrentUser.isCoOwnerOrOwner(eventProject) && !CurrentUser.isStaff()) {
+      // TODO: Handle someone other than owner
+    } else {
+      this.setState({
+        eventProject: eventProject,
+        startStep: url.argument("step") || 1,
+        steps: _.clone(this.state.steps),
+        isLoading: false,
+      });
+    }
+  }
+
+  handleLoadProjectError(error: APIError): void {
+    this.setState({
+      error: "Failed to load event project information",
+      isLoading: false,
+    });
+  }
+
+  onSubmit(
+    event: SyntheticEvent<HTMLFormElement>,
+    formRef: HTMLFormElement,
+    onSubmitSuccess: (EventProjectAPIDetails, () => void) => void
+  ): void {
+    const formSubmitUrl: string = `/api/event/${this.state.eventId}/projects/${this.state.projectId}/create/`;
+    api.postForm(
+      formSubmitUrl,
+      formRef,
+      onSubmitSuccess,
+      response => null /* TODO: Report error to user */
```

Comment:
todo, probably going to require toast implementation to put the error on the frontend the way we want?

Reply:
Recorded as "Display api errors on frontend"